Remove obsolete workaround for "most vexing parse" ambiguity.

MichaelGrupp opened the pull request:
The object initialization can't be misinterpreted as a function
declaration anymore since #353. The `+0` was a trick to avoid the most  
vexing parse problem in C++, which isn't needed anymore. Also removes
dummy classes that were probably used to experiment with said problem.

MichaelGrupp committed:
Remove some useless code from buffer_server_main.cpp

tfoote commented on tf2_ros/src/buffer_server_main.cpp:
```diff
@@ -76,13 +62,10 @@ int main(int argc, char** argv)
     node_name = "tf2_buffer_server";
   }
 
-  // WIM: this works fine:
```
How is this change related to the above unused code? How have you verified that this change is appropriate?

MichaelGrupp replied:
Yes, it works fine, see my long comment. I figured out why these things were added in the first place.

MichaelGrupp commented:
tl, dr: the code change works because the object initialization can't be misinterpreted as a function declaration anymore since https://github.com/ros/geometry2/pull/353

---
Long explanation:

Although the code comments were not clear (and the link is dead), it looks like the original intention of Wim was to avoid the "WTF" of the [most vexing parse](https://en.wikipedia.org/wiki/Most_vexing_parse). I use a vscode editor with cquery extension, which directly shows the `-Wvexing-parse` warning of Clang when I try to remove the "`+0`" from the original code (g++ shows a more cryptic warning):

![warning](https://user-images.githubusercontent.com/8985495/53559226-fc1e3400-3b49-11e9-979a-d23ff2f76038.png)

The `+0` was just a trick to force the compiler to not treat that line as a function declaration. An extra pair of parentheses would have the same effect. And the dummy classes that I deleted look very much like dummy classes used in examples of the most vexing parse problem :smile: 

Summary: 
```c++
// warning: parentheses were disambiguated as a function declaration [-Wvexing-parse]
tf2_ros::Buffer buffer_core(ros::Duration(buffer_size));

// unambiguous
tf2_ros::Buffer buffer_core(ros::Duration(buffer_size+0));

// since #353: unambiguous because of `publish_frame_service`, but useless `+0`
tf2_ros::Buffer buffer_core(ros::Duration(buffer_size+0), publish_frame_service));

// this PR: removed `+0`, still unambiguous
tf2_ros::Buffer buffer_core(ros::Duration(buffer_size), publish_frame_service));
```

MichaelGrupp commented:
Updated the PR description and title.

tfoote reviewed (approved):
Perfect, thanks for the detailed explanation.
